internal/host/statemachine.go

```diff
@@ -538,6 +542,8 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
 		},
 		Condition: stateswitch.And(If(IsConnected), If(HasInventory),
 			hasMinRequiredHardware,
+			hasMinHardwareForRole,
```

Comment:
This change is incorrect.  hasMinHardwareForRole should not be a condition for pending-for-input validation.

Reply:
That change is complementary to the new transition defined in https://github.com/openshift/assisted-service/pull/1738/files#diff-d26dc3c3c847de4b476ad1917cd7f5565935d06b6804e53d576ea04f26bc414bR574 and allows for the new transition to be chosen  instead of that one when it is desired.

 - The new transition enforces moving host to `Insufficient` when the hardware requirements for role are not met and `requiredInputFieldsExist` condition failed - that fixes the problem reported in the bug.
- The change you pointed out as incorrect makes sure that if requirements for hardware are met and the `requiredInputFieldsExist` is not, the host is moved to `pending-for-input`.

Does it make sense?

Reply:
The way these condition are arranged is:

1. First transition checks if basic hardware conditions exist https://github.com/openshift/assisted-service/blob/f5a57dc029e010df6aaf236797c800ab160d584e/internal/host/statemachine.go#L514
2. Second transition checks for user input  https://github.com/openshift/assisted-service/blob/f5a57dc029e010df6aaf236797c800ab160d584e/internal/host/statemachine.go#L530
3. Third transition checks all conditions that are required to be installed https://github.com/openshift/assisted-service/blob/f5a57dc029e010df6aaf236797c800ab160d584e/internal/host/statemachine.go#L549
4. Forth transition checks the same but all have to succeed (ready). https://github.com/openshift/assisted-service/blob/f5a57dc029e010df6aaf236797c800ab160d584e/internal/host/statemachine.go#L567

It seems to me that the operators conditions have to be added to be added together to _hasMinRequiredHardware_ and be handled across all transitions together. 

Anyway, since role is user input, role dependent conditions cannot be checked here before we verify the role.